src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,110 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  selectedPost: Post | null;
+  setCommentsOfPost: (comments: Comment[] | null
+  | ((prevVar: Comment[] | null) => Comment[] | null)) => void;
+  setIsErrorShowing: (value: boolean) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setCommentsOfPost,
+  setIsErrorShowing,
+}) => {
+  const [formInputs, setFormInputs] = useState({
+    author: '',
+    email: '',
+    text: '',
+  });
+  const [formErrors, setFormErrors] = useState({
+    author: false,
+    email: false,
+    text: false,
+  });
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleChangeInput = (
+    value: string,
+    field: string,
+  ): void => {
+    setFormInputs(prevForm => ({
+      ...prevForm,
+      [field]: value,
+    }));
+
+    setFormErrors(prevErrors => ({
+      ...prevErrors,
+      [field]: false,
+    }));
+  };
+
+  const handleReset = () => {
+    setFormInputs({
+      author: '',
+      email: '',
+      text: '',
+    });
```

Comment:
Create a default object as you do the same twice when passing as a default value, and here